diff -up audacious-plugins-4.1/build/src/qthotkey/meson.build.omv~ audacious-plugins-4.1/build/src/qthotkey/meson.build
diff -up audacious-plugins-4.1/src/aosd/meson.build.omv~ audacious-plugins-4.1/src/aosd/meson.build
--- audacious-plugins-4.1/src/aosd/meson.build.omv~	2022-01-22 18:48:44.701444242 +0100
+++ audacious-plugins-4.1/src/aosd/meson.build	2022-01-22 18:51:03.230852626 +0100
@@ -17,7 +17,7 @@ aosd_sources = [
 if have_aosd
   shared_module('aosd',
     aosd_sources,
-    dependencies: [audacious_dep, math_dep, gtk_dep, glib_dep, x11_dep],
+    dependencies: [audacious_dep, math_dep, gtk_dep, glib_dep, x11_dep, cxx.find_library('Xrender', required: true), cxx.find_library('Xcomposite', required: true)],
     name_prefix: '',
     install: true,
     install_dir: general_plugin_dir
diff -up audacious-plugins-4.1/src/glspectrum/meson.build.omv~ audacious-plugins-4.1/src/glspectrum/meson.build
--- audacious-plugins-4.1/src/glspectrum/meson.build.omv~	2022-01-22 18:49:11.773523943 +0100
+++ audacious-plugins-4.1/src/glspectrum/meson.build	2022-01-22 18:49:27.167569286 +0100
@@ -5,7 +5,7 @@ have_glspectrum = opengl_dep.found()
 if have_glspectrum
   shared_module('gl-spectrum',
     'gl-spectrum.cc',
-    dependencies: [audacious_dep, math_dep, gtk_dep, glib_dep, opengl_dep],
+    dependencies: [audacious_dep, math_dep, gtk_dep, glib_dep, opengl_dep, cxx.find_library('X11', required: true)],
     name_prefix: '',
     install: true,
     install_dir: visualization_plugin_dir
diff -up audacious-plugins-4.1/src/hotkey/meson.build.omv~ audacious-plugins-4.1/src/hotkey/meson.build
--- audacious-plugins-4.1/src/hotkey/meson.build.omv~	2022-01-22 18:48:03.830324030 +0100
+++ audacious-plugins-4.1/src/hotkey/meson.build	2022-01-22 18:48:21.239375213 +0100
@@ -12,7 +12,7 @@ hotkey_sources = [
 if have_hotkey
   shared_module('hotkey',
     hotkey_sources,
-    dependencies: [audacious_dep, audgui_dep, gtk_dep, glib_dep, gdk_x11_dep],
+    dependencies: [audacious_dep, audgui_dep, gtk_dep, glib_dep, gdk_x11_dep, cxx.find_library('X11', required: true)],
     name_prefix: '',
     install: true,
     install_dir: general_plugin_dir
diff -up audacious-plugins-4.1/src/qthotkey/meson.build.omv~ audacious-plugins-4.1/src/qthotkey/meson.build
--- audacious-plugins-4.1/src/qthotkey/meson.build.omv~	2022-01-22 18:37:00.219402710 +0100
+++ audacious-plugins-4.1/src/qthotkey/meson.build	2022-01-22 18:48:00.159313240 +0100
@@ -6,7 +6,7 @@ if have_qthotkey
   shared_module('qthotkey',
     'gui.cc',
     'plugin.cc',
-    dependencies: [audacious_dep, qt_dep, audqt_dep, qtx11extras_dep],
+    dependencies: [audacious_dep, qt_dep, audqt_dep, qtx11extras_dep, cxx.find_library('X11', required: true) ],
     name_prefix: '',
     install: true,
     install_dir: general_plugin_dir
